Modules/_zstd/compressor.c

```diff
@@ -195,7 +196,7 @@ _get_CDict(ZstdDict *self, int compressionLevel)
             goto error;
         }
 
-        /* Add PyCapsule object to self->c_dicts */
+        /* Add PyCapsule object to self->c_dicts if not already inserted */
```

Comment:
I think you should use `PyDict_SetDefaultRef` instead of `PyDict_SetItem` below

Reply:
Done! I didn't realize the semantics around SetItem were quite so weird.

Reply:
Ah, no worries! I don't think using SetDefaultRef is wrong per se. But I can always undo that change.

Reply:
Yeah, I don't think it's wrong, just maybe more complicated than necessary. Up to you.

Reply:
I think I will revert. Do you have thoughts on using PyDict_GetItemRef here: https://github.com/python/cpython/pull/134289#discussion_r2102566026 ?

I would think similarly it's not really needed.